### PR TITLE
Use cibuildwheel

### PR DESCRIPTION
**Pull request by joaander**
## Description

Use cibuildwheel to build wheels.

## Motivation and Context

Add wheels for MacOS.

## How Has This Been Tested?

The CI tests themselves are the test.

## Change log

```
Added:

* PyPI wheels for macos-x86_64 and macos-arm64.

Changed:

* Skip large file tests by default.
```

## Checklist:

- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/gsd/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**GSD Contributor Agreement**](https://github.com/glotzerlab/gsd/blob/master/ContributorAgreement.md).
- [x] My name is on the list of contributors (`doc/credits.rst`) in the pull request source branch.

### CODE REVIEW

**Comment by joaander** [comment]
@bdice @vyasr Tagging you since you are interested. The changes to use `cibuildwheel` and fully support PEP517 are in `.github/workflows/build_wheels.yml` and `pyproject.toml`. The other changes adjust how I activate the longer running tests (pytest's --ignore option is rather flaky).

I definitely think that `cibuildwheel` is worth using for freud - at some future date when time permits (I could try the conversion if you would like). `cibuildwheel` has more options than I use in gsd. In freud,  you'll need `CIBW_BEFORE_ALL` to install tbb: https://cibuildwheel.readthedocs.io/en/stable/options/#before-all.

**Comment by vyasr** [comment]
Thanks Josh! I'll probably take a look at it this weekend.

File: pyproject.toml [hunk 1]
```diff
@@ -1,2 +1,8 @@
+[project]
+requires-python = ">=3.6"
+
 [build-system]
-requires = ["setuptools", "wheel", "cython", "numpy"]
+requires = ["setuptools>=42",
+            "wheel",
+            "Cython",
+            "oldest-supported-numpy"]
```
Comment (bdice) [inline pyproject.toml:8]: I'm unclear on whether `oldest-supported-numpy` is appropriate here. What happens if you disable PEP517?

File: .github/workflows/build_wheels.yml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    name: Build wheels on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-20.04, windows-2019, macos-10.15]
+
+    steps:
+      - uses: actions/checkout@v2.3.4
+
+      - name: Install Visual C++ for Python 2.7
```
Comment (bdice) [inline .github/workflows/build_wheels.yml:33]: I think the Python 2.7 steps could be deleted.

**Review by bdice — APPROVED** [review]
LGTM, but I have two questions. @joaander If you want to try to enable this for freud, go right ahead!

File: pyproject.toml (hunk 1 above)
Comment (joaander) [reply pyproject.toml:8]: Builds that disable PEP517 use `setup.py` is used directly and skip `pyproject.toml`. GSD's `setup.py` imports numpy, so this style build requires that users manually install numpy first.

The `oldest-supported-numpy` [README](https://github.com/scipy/oldest-supported-numpy) answers some FAQs, like "Can I use this if my package requires a recent version of Numpy?" It points out that `pyproject.toml` describes the build system itself, while `setup.py` defines the installation requirements. While a project might use newer numpy python API calls, those may not be needed to build against the numpy C API.

File: .github/workflows/build_wheels.yml (hunk 2 above)
Comment (joaander) [reply .github/workflows/build_wheels.yml]: You are right.

**Pull request merged**